test/python/transpiler/test_pass_scheduler.py

```diff
@@ -609,12 +591,14 @@ def test_nested_conditional_in_loop(self):
 class DoXTimesController(FlowController):
     """A control-flow plugin for running a set of passes an X amount of times."""
 
-    def __init__(self, passes, options, do_x_times=0, **_):
-        self.do_x_times = do_x_times()
-        super().__init__(passes, options)
+    def __init__(self, passes, options, do_x_times, **_):
+        super().__init__(options)
+        self.passes = passes
+        self.do_x_times = do_x_times
 
-    def __iter__(self):
-        for _ in range(self.do_x_times):
+    # pylint: disable=missing-function-docstring
+    def iter_tasks(self, property_set):
+        for _ in range(self.do_x_times(property_set)):
             yield from self.passes
```

Comment:
Hmm, I'm thinking for backwards compatibility, perhaps we should give `BaseController.iter_tasks` a default implementation that just does `yield from iter(self)`, but still say that subclasses should override it?

Reply:
Hmm writing `iter(self)` without implementing `__iter__` (because structure of pipeline is now subclass dependent) looks bit weird to me.